js/components/adPanel.js

```diff
@@ -14,6 +14,13 @@ var React = require('react'),
     Utils = require('./utils'),
     Icon = require('../components/icon');
 
+var state = { 
```

Comment:
@ipadilla4 the convention here seems to be to use this.setState(), see for example in discoveryPanel.js. @aeng7 can you confirm that is correct

Reply:
Agreed, we should use React's component state whenever possible.

Reply:
Note that some of these can potentially be props.

Reply:
@ipadilla4 can you investigate converting some of these into props to address the infinite loop issue you mentioned earlier?

Reply:
Ok I used props before but I was having issues with object persistence, maybe I was using the wrong "upper" component. I will try this approach.